### DISCUSSION TITLE
Unable to access Web Console using Getting Started Guide

### DISCUSSION THREAD

**Discussion by Adam2Marsh** [arkmq-org/arkmq-org-broker-operator#427] [Q&A] [answered]
Morning All,

We've just followed the Getting Started Guide on deploying our first AMQ instance using the operator. That all went fine until we try to access the Web Console via kubectl poer-forward.

```
~/P/s/g/u/k/kubernetes-cluster-state-non-prod ❯❯❯ k port-forward ex-aao-ss-0 8161:8161                  ✘ 130 master
Forwarding from 127.0.0.1:8161 -> 8161
Forwarding from [::1]:8161 -> 8161
Handling connection for 8161
Handling connection for 8161
E1221 10:38:31.998728   49577 portforward.go:406] an error occurred forwarding 8161 -> 8161: error forwarding port 8161 to pod 42cf6b0ff733f61bc333b10c81d6f285da309530445b0a8d4cae7505e317cfc7, uid : failed to execute portforward in network namespace "/var/run/netns/cni-e7044645-06df-c978-e6ed-0a33e70b2dbf": failed to connect to localhost:8161 inside namespace "42cf6b0ff733f61bc333b10c81d6f285da309530445b0a8d4cae7505e317cfc7", IPv4: dial tcp4 127.0.0.1:8161: connect: connection refused IPv6 dial tcp6: address localhost: no suitable address found 
E1221 10:38:31.998905   49577 portforward.go:406] an error occurred forwarding 8161 -> 8161: error forwarding port 8161 to pod 42cf6b0ff733f61bc333b10c81d6f285da309530445b0a8d4cae7505e317cfc7, uid : failed to execute portforward in network namespace "/var/run/netns/cni-e7044645-06df-c978-e6ed-0a33e70b2dbf": failed to connect to localhost:8161 inside namespace "42cf6b0ff733f61bc333b10c81d6f285da309530445b0a8d4cae7505e317cfc7", IPv4: dial tcp4 127.0.0.1:8161: connect: connection refused IPv6 dial tcp6: address localhost: no suitable address found 
E1221 10:38:32.002408   49577 portforward.go:234] lost connection to pod
Handling connection for 8161
E1221 10:38:32.005018   49577 portforward.go:346] error creating error stream for port 8161 -> 8161: EOF
```

I've not been able to find any documentation to advise further configurations is required to make the conole availablel; according to the ActiveMQ Artemis docs it should be there by default - https://activemq.apache.org/components/artemis/documentation/2.21.0/management-console.html

When digging around I see 2 problems;

1. The pod does not have any ports configured to expose
2. The bootstrap.xml is using the hostname of the pod to expose the management console; I found the following issues from over a year ago with no answers - https://github.com/artemiscloud/activemq-artemis-operator/issues/96

Am I doing something silly? Have I missed something? Can anyone offer some advice as I can't be the only person to of had this problem.

We're using the latest version of the operator and deploying an AMQ instance with the following:
```
apiVersion: broker.amq.io/v1beta1
kind: ActiveMQArtemis
metadata:
  name: ex-aao
  namespace: uk-landline-sandbox
spec:
  adminUser: admin
  adminPassword: password
  deploymentPlan:
    size: 1
    image: placeholder
  console:
    expose: true
```

**Comment by Adam2Marsh** [comment]
We can now access but we had to go via ingress.

Is this expected? Is there a reason we can't just use port-forwarding with the server running the web console on 0.0.0.0 inside the container.

**Comment by brusdev** [comment, marked as answer]
The #385 PR will add a new container port for the console but the embedded web server only listens the internal IP address for security reasons,  the only way to access to the console is via ingress/route setting `console.expose` to true.
